docs/aws.md

```diff
@@ -0,0 +1,76 @@
+    2. Go to the IAM console.
+    3. In the navigation pane, choose Policies.
+    4. Create a new policy by clicking "Create Policy".
+    5. Next, click JSON and paste the following JSON with the required permissions and then click "Next".
+      ```json
+    {
+      "Version": "2012-10-17",
+      "Statement": [
+        {
+          "Effect": "Allow",
+          "Action": [
+            "ec2:RunInstances",
+            "ec2:TerminateInstances",
+            "ec2:DescribeInstances",
+            "ec2:DescribeInstanceStatus"
+          ],
+          "Resource": "*"
+        }
+      ]
+    }
+      ```
+    6. Name the policy (for example: `gha-runner-policy`) and select "Create Policy".
+    7. Now select "Users" in the navigation pane and click "Add user".
+    8. Enter a username (for example: `gha-runner-user`) and ensure that "Provide user access to the AWS Management Console" is unchecked.
+    9. Select "Attach policies directly" and search for the policy you created earlier (`gha-runner-policy`) and select it.
+    10. Then click, "Create user".
+    11. Next, click on the user you just created and go to the "Security credentials" tab.
+    12. Click "Create access key" and then click "Other" and then "Next".
+    13. Click next and then copy the Access Key and Secret Access Key (or download the CSV file), these will not be shown again.
+2. Create your GitHub Access Token
+    1. This can be done with either a Personal Access Token or a Fine-Grained Personal Access Token.
+    2. Go to your GitHub account settings.
+    3. Click on "Developer settings".
+    4. Create a new token with `repo` scope.
+    5. Save and/or copy the token.
+3. Add your credentials to your repository secrets
+    1. Go to your repository on GitHub.
+    2. Click on "Settings", then "Secrets and Variables", and then "Actions".
+    3. Click "New repository secret".
+    4. Add the following secrets:
+      - `AWS_ACCESS_KEY_ID` - The Access Key you copied earlier.
+      - `AWS_SECRET_ACCESS_KEY` - The Secret Access Key you copied earlier.
+      - `GITHUB_TOKEN` - The GitHub token you copied earlier.
+4. Prepare an EC2 image
+    1. Go to the AWS Management Console and sign in to your account.
+    2. Go to the EC2 console.
+    3. Click "Launch Instance".
+    4. Select an Amazon Linux 2 AMI.
+    5. Choose an instance type.
+    6. Configure the instance details.
+    7. Connect to the instance and install `docker`, `git`, and enable the `docker` service. This will be distro dependent.
+    8. Feel free to install any other useful tools that you may need to get started.
+    9. Create an image of the instance using the following [AWS Docs](https://docs.aws.amazon.com/toolkit-for-visual-studio/latest/user-guide/tkv-create-ami-from-instance.html) (this may take a while)
+    10. Make sure to remove the instance once you are done creating the image.
+5. Optional: Create a VPC with subnet and security group.
```

Comment:
It might be worth explaining why you would/wouldn't need a VPC/security group. Am I correct that the main reason to specify the VPC is if you're using the AWS account for other things, and was to keep a separate VPC for different account use cases?

Reply:
You are correct on a VPC. Your security group is more focused on access and what machines might be able to SSH in your machines. I go back and forth on whether or not the VPC/subnet docs should be included because most of our community doesn't have a need for it right now. Those who have a need, generally know they need to set it up.

Reply:
These have been removed in [dcf41b5](https://github.com/omsf-eco-infra/gha-runner/pull/16/commits/dcf41b5727a76136d13603af865d309c5bc044e4)